resources/migrations/001_update_migrations.yaml

```diff
@@ -11957,6 +11957,40 @@ databaseChangeLog:
                            'v54.2025-02-14T08:05:00'
                   );
 
+  - changeSet:
+      id: v55.2025-06-22T10:00:00
+      author: ranquild
+      comment: |
+        Update `has_field_values` on FK fields to match PKs. Before this change, the FE always used the corresponding
+        PKs fields for FKs fields in dashboards. Now the FE does not swap fields, and we need to set `has_field_values`
+        to prevent any changes in behavior.
+      changes:
+        - sql:
+            dbms: postgresql
+            sql: >-
+              UPDATE metabase_field AS fk
+              SET has_field_values = pk.has_field_values
+              FROM metabase_field AS pk
+              WHERE pk.id = fk.fk_target_field_id AND pk.has_field_values IS NOT NULL AND fk.semantic_type = 'type/FK';
+        - sql:
+            dbms: mysql,mariadb
+            sql: >-
+              UPDATE metabase_field AS fk
+              JOIN metabase_field AS pk ON pk.id = fk.fk_target_field_id
+              SET fk.has_field_values = pk.has_field_values
+              WHERE pk.has_field_values IS NOT NULL AND fk.semantic_type = 'type/FK';
+        - sql:
+            dbms: h2
+            sql: >-
+              UPDATE metabase_field AS fk
+              SET has_field_values = (
+                SELECT coalesce(pk.has_field_values, fk.has_field_values)
+                FROM metabase_field AS pk
+                WHERE pk.id = fk.fk_target_field_id
+              )
+              WHERE fk_target_field_id IS NOT NULL AND semantic_type = 'type/FK';
```

Comment:
Have you considered a cross platform query like
```sql
UPDATE metabase_field 
   SET has_field_values = (
       SELECT pk.has_field_values 
       FROM metabase_field pk 
       WHERE pk.id = metabase_field.fk_target_field_id
   )
 WHERE semantic_type = 'type/FK'
   AND has_field_values IS DISTINCT FROM (
       SELECT pk.has_field_values 
       FROM metabase_field pk 
       WHERE pk.id = metabase_field.fk_target_field_id
   );
```

Not doing the update when the two values are the same might be beneficial.